lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -72,6 +73,22 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+        user_id = usage_info.user.id if usage_info.user else None
```

Comment:
`%%USER_ID%%` is supposed to be replaced with an anonymous user id, not the actual user ID, so you'll need to use `anonymous_id_for_user(user, course)` to get the anonymous ID.

Reply:
Addressed by https://github.com/edx/edx-platform/pull/15905/commits/86235d261e710f16a4790a761e21b604e4883342#diff-1a30c64335d7765247aec974a4d8601eR91